Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -82,7 +170,26 @@ ThrowCompletionOr<Object*> RegExpConstructor::construct(FunctionObject&)
         flags_value = flags;
     }
 
-    return TRY(regexp_create(vm, pattern_value, flags_value));
+    // https://github.com/tc39/proposal-regexp-legacy-features#regexpalloc--newtarget-
```

Comment:
Can we implement this as a standalone `regexp_alloc()` function instead that is then being called here as per https://tc39.es/ecma262/#sec-regexp-pattern-flags?